budgetr-backend/src/test/java/pl/mantiscrab/budgetr/domain/bankaccount/InMemoryBankAccountRepository.java

```diff
@@ -0,0 +1,49 @@
+package pl.mantiscrab.budgetr.domain.bankaccount;
+
+import pl.mantiscrab.budgetr.InMemoryDummyCrudRepository;
+import pl.mantiscrab.budgetr.domain.user.User;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+class InMemoryBankAccountRepository extends InMemoryDummyCrudRepository<BankAccount, Long> implements BankAccountRepository {
+    @Override
+    public Optional<BankAccount> findById(Long id) {
+        BankAccount bankAccount = map.get(id);
+        return Optional.ofNullable(bankAccount);
+    }
+
+    @Override
+    public <S extends BankAccount> S save(S bankAccount) {
+        if (bankAccount.getId() == null) {
+            bankAccount.setId(index);
+            index++;
+        }
+        map.put(bankAccount.getId(), bankAccount);
+        return bankAccount;
+    }
+
+    @Override
+    public List<BankAccount> findByUser(User user) {
+        return map.values().parallelStream()
```

Comment:
What gives us a parallel stream here? Do you have some test for that?

Reply:
I simply makes filtering a little bit faster in here, but I will change it since you pointed it.